shm/player.cpp

```diff
@@ -0,0 +1,48 @@
+#include "player.hpp"
+
+#include <numeric>
+
+Player::Player(std::shared_ptr<Ship> ship, uint32_t money, uint32_t availableSpace)
+    : ship_(ship), money_(money), availableSpace_(availableSpace) {}
+
+std::shared_ptr<Ship> Player::getShip() {
+    return ship_;
+}
+uint32_t Player::getAvailableSpace() const {
+    return availableSpace_;
+}
+uint32_t Player::getMoney() const {
+    return money_;
+}
+
+uint32_t Player::CountAvailableSpace() const {
+    uint32_t sumOfAmounts =
+        std::accumulate(ship_->getVectorCargo().begin(), ship_->getVectorCargo().end(), 0,
+                        [](size_t amounts, const auto& cargo) { return amounts += cargo.get()->getAmount(); });
```

Comment:
Przy kompilowaniu miałem problem z tym "cargo.get()" wiesz o co może chodzić?
```/home/kasper/Projekty/coders_school/object-oriented-programming/shm/player.cpp:23:89: error: ‘const class Cargo’ has no member named ‘get’
   23 |                         [](size_t amounts, const auto& cargo) { return amounts += cargo.get()->getAmount(); });```

Reply:
Pytanie czy w klasie Ship zmieniłeś na typ z cargo* na shared_ptr<cargo>. cargo.get() to wyłuskanie z shared pointera.

Reply:
Sprawdź tutaj jak możesz: https://github.com/Kasperice/object-oriented-programming/tree/makefile tam robiłem brancha na podstawie shm2 (możliwe że tam czegoś jeszcze brakuje)